### DISCUSSION TITLE
How can I group multiple roles into one role?

### DISCUSSION THREAD

**Discussion by Restingo** [spatie/laravel-permission#1877] [Q&A]
Hi,

I want to ask how I can group multiple roles into one? For example I've got multiple reports in my application like:

- Sales (sales.create, sales.update, ...)
- Purchases (purchases.create, purchases.update, ...)

and I got roles like

- Salesadministrator (all sales permissions)
- Purchasesadministrator (all purchase permissions)

and I want to group them in a role that is named like

- Accountant (Salesadministrator, Purchasesadministrator)

How can I do this? I already added the `HasRoles` trait to Roles model but it doesn't work. Thank you.

**Comment by erikn69** [comment]
> I already added the `HasRoles` trait to Roles model but it doesn't work

Did you add the `role_id` on the Roles as foreing key?

You don't need `HasRoles`, on Roles you only need
```php
public function roles() {
    return $this->hasMany(
        config('permission.models.role'),
        'role_id',
        'id'
    );
)
```
Now when you assign, find all roles with childs
```php
$ids=[];
Roles::with('roles')->whereIn($ids_to_assign)->get()->each(function($role)use(&$ids){
    $ids[]=$role->id;
    $ids=array_merge($ids,$role->roles->pluck('id')->toArray());
});
$user->syncRoles(array_unique($ids));
```

**Reply by Restingo** [reply]
Hi @erikn69,

thanks for your fast reply. Would it be possible to keep the roles instead of looking them up and assigning them directly? I want to clean up my whole role construct with this as I have many roles. So that I can assign only the role Accountant to my user and the user automatically has all permissions of the child roles but if I look up the roles of the user it only gives me Accountant?

**Reply by erikn69** [reply]
Maybe overriding `hasRole` or overriding cache remember adding group roles
